tests/test_ledger.py

```diff
@@ -316,4 +346,35 @@ def test_mirror_ledger(add_vat_code):
     # Restore initial state
     cashctrl_ledger.mirror_ledger(target=initial, delete=True)
     mirrored = cashctrl_ledger.ledger().reset_index(drop=True)
-    assert txn_to_str(initial) == txn_to_str(mirrored)
\ No newline at end of file
+    assert txn_to_str(initial) == txn_to_str(mirrored)
+
+def test_get_ledger_attachments(files):
+    cashctrl_ledger = CashCtrlLedger()
+
+    # Mirror ledger entries
+    target = pd.concat([individual_transaction, collective_transaction])
+    initial_ledger = cashctrl_ledger.ledger().reset_index(drop=True)
+    cashctrl_ledger.mirror_ledger(target=target)
+    mirrored = cashctrl_ledger.ledger().reset_index(drop=True)
+    outer_join = pd.merge(initial_ledger, mirrored, how='outer', indicator=True)
+    created_ledger = outer_join[outer_join['_merge'] == "right_only"].drop('_merge', axis = 1)
+    document_dict = created_ledger.set_index('id')['document'].to_dict()
+
+    # Manually attach files
+    for id, document in document_dict.items():
+        file_id = cashctrl_ledger._client.file_path_to_id(document)
+        cashctrl_ledger._client.post("journal/update_attachments.json", data={'id': id, 'fileIds': file_id})
+    cashctrl_ledger._client.invalidate_journal_cache()
```

Comment:
I find this setup overly complex. 
Why do we need to know about mirroring, individual_transactions, and collective_transactions defined above?

Here's a sketch for a simpler test setup:
(In addition, teardown is more robust thanks to in fixtures.)

```python

@pytest.fixture
def ledger_ids():
    """Populate remote ledger with three new entries and return their ids in a list."""
    entry = pd.DataFrame({
        'date': ['2024-05-24'],
        'account': [2270],
        'counter_account': [2210],
        'amount': [100],
        'currency': ['CHF'],
        'text': ['test entry'],
    }
    engine = CashCtrlLedger()
    ledger_ids = [engine.add_ledger_entry(entry) for _ in range(3)]

    yield ledger_ids

    # Restore original ledger state
    engine.delete_ledger(ledger_ids)

def test_get_ledger_attachments(files, ledger_ids):
    engine = CashCtrlLedger()
    initial = engine._get_ledger_attachments() == {}

    engine._client.post("journal/update_attachments.json", data={'id': ledger_ids[0], 'fileIds': files['id'].iat[0]})
    expected = initial | {ledger_ids[0]: ['/file1.txt']}
    assert engine._get_ledger_attachments() == expected

    engine._client.post("journal/update_attachments.json", data={'id': ledger_ids[1], 'fileIds': files['id'].iat[1]})
    expected = initial | {ledger_ids[0]: ['/file1.txt'], ledger_ids[1]: ['/subdir/file2.txt']}
    assert engine._get_ledger_attachments() == expected

    engine._client.post("journal/update_attachments.json", data={'id': ledger_ids[1], 'fileIds': files['id'].iat[0]})
    expected = initial | {ledger_ids[0]: ['/file1.txt'], ledger_ids[1]: ['/subdir/file2.txt', '/file1.txt']}
    assert engine._get_ledger_attachments() == expected
```

The proposal requires extending `add_ledger_entry` to return the id of the created entry (which sounds like a good idea anyhow).